nixos/modules/services/web-apps/mastodon.nix

```diff
@@ -708,13 +711,27 @@ in {
       script = ''
         umask 077
 
+        if ! test -f /var/lib/mastodon/secrets/active-record-encryption-deterministic-key; then
+          mkdir -p /var/lib/mastodon/secrets
+          bin/rails db:encryption:init | grep --only-matching "ACTIVE_RECORD_ENCRYPTION_DETERMINISTIC_KEY=[^ ]\+" | sed 's/^ACTIVE_RECORD_ENCRYPTION_DETERMINISTIC_KEY=//' > /var/lib/mastodon/secrets/active-record-encryption-deterministic-key
+        fi
+
+        if ! test -f /var/lib/mastodon/secrets/active-record-encryption-key-derivation-salt; then
+          mkdir -p /var/lib/mastodon/secrets
+          bin/rails db:encryption:init | grep --only-matching "ACTIVE_RECORD_ENCRYPTION_KEY_DERIVATION_SALT=[^ ]\+" | sed 's/^ACTIVE_RECORD_ENCRYPTION_KEY_DERIVATION_SALT=//' > /var/lib/mastodon/secrets/active-record-encryption-key-derivation-salt
+        fi
+
+        if ! test -f /var/lib/mastodon/secrets/active-record-encryption-primary-key; then
+          mkdir -p /var/lib/mastodon/secrets
+          bin/rails db:encryption:init | grep --only-matching "ACTIVE_RECORD_ENCRYPTION_PRIMARY_KEY=[^ ]\+" | sed 's/^ACTIVE_RECORD_ENCRYPTION_PRIMARY_KEY=//' > /var/lib/mastodon/secrets/active-record-encryption-deterministic-key
```

Comment:
Is it safe to run this as three separate invocations?

Reply:
Yeah, the keys are independent of each other.
I haven't finished this piece of code yet.